axelrod/match.py

```diff
@@ -77,11 +82,28 @@ def play(self):
 
         i.e. One entry per turn containing a pair of actions.
         """
+        if self._prob_end is not None:
```

Comment:
This sampling is statistically equivalent to a flip of a coin at the end of
each pair of actions. Doing it this way just saves from repeatedly
'flipping a coin'.

I think it's better for the randomization to happen at a match level. It
makes things easier further down the line (I think) and also is closer to
the idea of my interpretation of a stochastic length game. I spent some
time thinking about this and I think that once Owen has finished what he's
working on things will be easier to modularize.

I'm mainly opening this PR to get people's thoughts as I'm not convinced
mine are correct so I'm not at all immovable here.

I don't think we want to change the value of turns. Mainly because of some
cheating strategies... I thought about that for a bit but felt it was
better to keep things separate?

On Thu, 18 Feb 2016, 20:47 Marc Harper <notifications@github.com> wrote:

> In axelrod/match.py
> <https://github.com/Axelrod-Python/Axelrod/pull/461#discussion_r53379806>:
>
> > @@ -77,11 +82,28 @@ def play(self):
> >
> >          i.e. One entry per turn containing a pair of actions.
> >          """
> > +        if self._prob_end is not None:
>
> It feels like the statistical details here are not really the concern of
> the Match class -- it just needs to know the value of end_turn, not how
> to generate it. Perhaps it's better to factor this part out (which also has
> the advantage of allowing other ending schemes), and simply pass end_turn
> to Match.__init__.
>
> OTOH perhaps it's better for the calling code to change the value of turns
> (handling the randomization outside the Match class). I guess there are
> several ways one might want to do this...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/461/files#r53379806>.
>